Dfe.Data.SearchPrototype/Web/Tests/ViewTests/PaginationPresentationTests.cs

```diff
@@ -0,0 +1,378 @@
+        IHtmlListItemElement? previousEllipsisListItem = paginationListItems[2] as IHtmlListItemElement;
+        previousEllipsisListItem?.ClassName.Should().NotContain("govuk-visually-hidden");
+
+        // button 19 
+        IHtmlListItemElement? pageNumber19ListItem = paginationListItems[3] as IHtmlListItemElement;
+        IElement? page19Button = pageNumber19ListItem?.Children.First(element => element.Id == "pageNumber-19");
+        page19Button?.TextContent.Should().Contain("19");
+
+        // button 20
+        IHtmlListItemElement? pageNumber20ListItem = paginationListItems[4] as IHtmlListItemElement;
+        IElement? page20Button = pageNumber20ListItem?.Children.First(element => element.Id == "pageNumber-20");
+        page20Button?.TextContent.Should().Contain("20");
+
+        // button 21, current page 
+        IHtmlListItemElement? pageNumber21ListItem = paginationListItems[5] as IHtmlListItemElement;
+        pageNumber21ListItem?.ClassName.Should().Contain("govuk-pagination__item--current");
+        IElement? currentPageLabelElement = pageNumber21ListItem?.Children.First(element => element.Id == $"pageNumber-{CurrentPageNumber}");
+        currentPageLabelElement?.TextContent.Should().Contain($"{CurrentPageNumber}");
+
+        // button 22
+        IHtmlListItemElement? pageNumber22ListItem = paginationListItems[6] as IHtmlListItemElement;
+        IElement? page22Button = pageNumber22ListItem?.Children.First(element => element.Id == "pageNumber-22");
+        page22Button?.TextContent.Should().Contain("22");
+
+        // button 23
+        IHtmlListItemElement? pageNumber23ListItem = paginationListItems[7] as IHtmlListItemElement;
+        IElement? page23Button = pageNumber23ListItem?.Children.First(element => element.Id == "pageNumber-23");
+        page23Button?.TextContent.Should().Contain("23");
+
+        // next ellipsis
+        IHtmlListItemElement? nextEllipsisListItem = paginationListItems[8] as IHtmlListItemElement;
+        nextEllipsisListItem?.ClassName.Should().NotContain("govuk-visually-hidden");
+
+        // last page number , total page number
+        IHtmlListItemElement? lastPageNumberListItem = paginationListItems[9] as IHtmlListItemElement;
+        lastPageNumberListItem?.ClassName.Should().NotContain("govuk-visually-hidden");
+        IElement? lastPageButton = lastPageNumberListItem?.Children.First(element => element.Id == "pageNumber-31");
+        lastPageButton?.TextContent.Should().Contain("31");
+
+        //next button
+        IHtmlListItemElement? nextPageListItem = paginationListItems[10] as IHtmlListItemElement;
+        nextPageListItem?.ClassName.Should().NotContain("govuk-visually-hidden");
+        IElement? nextPageButton = nextPageListItem?.Children.First(element => element.Id == "next-page");
+        nextPageButton?.TextContent.Should().Contain("next");
+    }
+
+    [Fact]
+    public async Task Pagination_WithCurrentPageInsideUpperBoundry_ShowsCorrectPaginationElements()
```

Comment:
typo **boundry**

Reply:
@spanersoraferty I told you I didn't like this word! (I changed it, google says its **boundary**)